src/types/env/mod.rs

```diff
@@ -0,0 +1,25 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Email provider type.
+
+#[cfg(test)]
+mod test;
+
+use serde::de::Error;
+
+use types::error::{AppError, AppErrorKind};
+
+enum_boilerplate!(Env ("env", InvalidEnv) {
+    Dev => "dev",
+    Prod => "production",
+    Stage => "staging",
+    Test => "test",
+});
+
+impl Default for Env {
```

Comment:
Ultimately we may move this implementation of `Default` into `enum_boilerplate!` too. I didn't do it yet, because I wasn't sure if we'll have a suitable implementation for everything that uses the macro. We can move it in later if it turns out we do.